akka-stream-testkit/src/test/scala/akka/stream/testkit/StreamSpec.scala

```diff
@@ -54,7 +65,17 @@ abstract class StreamSpec(_system: ActorSystem) extends AkkaSpec(_system) {
                 akka.stream.testkit.scaladsl.StreamTestKit.snapshotString(s.asInstanceOf[StreamSnapshotImpl])))
         }
         failed
+      case other if _assertChildren =>
+        Materializer(_system) match {
+          case impl: PhasedFusingActorMaterializer =>
+            stopAllChildren(impl.system, impl.supervisor)
+            val result = test.apply()
+            assertNoChildren(impl.system, impl.supervisor)
+            result
+          case _ => other
+        }
       case other => other
+
```

Comment:
Hmm, so we didn't actually have a check here as I thought in #27743 only something on failure trying to find stuck stages (which looks like it may not be correct anymore).

I think we should skip the `_assertChildren` flag change and instead make `MaterializerStateSpec` not extending `StreamSpec`, it is anyway a bit different in that it creates materializers and asserts things about them itself rather than testing some stream.

Reply:
Yes, it seems better than now.